### DISCUSSION TITLE
Do I need to sign CLA or sign off my commits to contribute as a member of a company

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [apache/airflow#44578] [Q&A] [answered]
Hello. I hope your day is going well. I'm working at Verkada and considering to contribute using company's time and equipment. Our legal is asking if there's a need to sign CLA, or I wonder if there's something similar, like signing off commits. I haven't found any requirements like that in the contributors guide, but wonder if I missed something. Can you confirm it or show what I've missed, please?

**Comment by potiuk** [comment, marked as answer]
If you get to the point that we invite you as committer (https://github.com/apache/airflow/blob/main/COMMITTERS.rst) then one of the conditions is to sign ICLA: https://www.apache.org/licenses/contributor-agreements.html - no need if you do casual contributions, but you are encouraged to do so regardless. Feel free to do it  any time following that link above.

You do not need permission from your employer to do so - everything you do for ASF projects is in your individual capacity, not your employer. But if your employer wants - they can also sign CCLA (also see the link above) and mention you there - it's totally optional, but some of the companies want to have such CCLA signed.
